### PR TITLE
Serveral Minor fixes 

### PR DESCRIPTION
**Pull request by girishmg**
This PR includes several minor fixes:

- miscellaneous fixes to handle unused parameter and unhandled errors
- MonitorCancel API only needs jsonContext to match the previous monitor
- fix unhandled error in Disconnect()
- call to 'err.Error()' may lead to nil pointer dereference in Connect

@hzhou8 @noah8713 @vtolstov PTAL

### CODE REVIEW

File: client.go [hunk 1]
```diff
@@ -183,7 +183,7 @@ func echo(client *rpc2.Client, args []interface{}, reply *[]interface{}) error {
 
 // RFC 7047 : Update Notification Section 4.1.6
 // Processing "params": [<json-value>, <table-updates>]
-func update(client *rpc2.Client, params []interface{}, reply *interface{}) error {
+func update(client *rpc2.Client, params []interface{}, _ *interface{}) error {
```
Comment (vtolstov) [inline client.go:186]: does it really needed to have func with ignored arg?

File: client.go [hunk 2]
```diff
@@ -356,5 +356,5 @@ func handleDisconnectNotification(c *rpc2.Client) {
 
 // Disconnect will close the OVSDB connection
 func (ovs OvsdbClient) Disconnect() {
-	ovs.rpcClient.Close()
+	_ = ovs.rpcClient.Close()
```
Comment (vtolstov) [inline client.go]: may be return error in case of error?

**Review by vtolstov — CHANGES_REQUESTED** [review]

File: client.go (hunk 1 above)
Comment (girishmg) [reply client.go:186]: the `github.com/cenkalti/rpc2` module that calls the `update` callback function on receiving the `update` method from the OVSDB-Server expects all the callback function to have two arguments. That library doesn't know that the `update` is one-way call and doesn't expect reply.

With `reply` not changed to `_`, it is getting tagged as error in my ide, so the common convention that I have seen is to set it '_'.

**Review by hzhou8 — APPROVED** [review]

File: client.go (hunk 2 above)
Comment (girishmg) [reply client.go]: @vtolstov the fix in this PR is to just silence my IDE in complaining whenever I try to `commit` changes. I will submit a separate PR to handle the error case. Thanks.

So, can you please accept the PR and then I will submit a new one? Thanks.

**Comment by girishmg** [comment]
@vtolstov @hzhou8 can this be merged please? thanks

File: client.go (hunk 2 above)
Comment (vtolstov) [reply client.go]: as the fix is simple, i'm prefer to get it with this pr. or if you don't have time - revert changes that silence vet tools

**Review by vtolstov — CHANGES_REQUESTED** [review]

File: client.go (hunk 2 above)
Comment (girishmg) [reply client.go]: @vtolstov this PR was not meant for that, so I am not sure what is the resistance to do this in a phased manner.

**Comment by girishmg** [comment]
@vtolstov I removed that commit which made the change to `Disconnect()`

I will address that change to `Disconnect` in an another commit and PR.

**Comment by hzhou8** [comment]
@vtolstov looks like your comments were addressed, so I merged it.

**Pull request merged**